sale_channel_hook/models/sale_channel.py

```diff
@@ -19,6 +16,10 @@ class SaleChannel(models.Model):
     auth_token = fields.Char("Secret authentication token")
     api_endpoint = fields.Char("Hooks API endpoint")
 
+    def _apply_webhook_security(self, headers, payload, url):
```

Comment:
The idea is not to remove all the auth code but to implement
- a select field on the channel to choose the auth method (signature ou basic_auth)
- implement the 2 method _auth_basic, auth_signature
- in the method _apply_webhook_security (maybe better to name it _add_weebhook_auth) call the right method depending of the method selected

Reply:
Done. Following our discussion about testing requests, I just used tests for the header/content/url of the request and use json=[...], it seems inappropriate to use mock_requests here